lib/rubocop/cop/style/redundant_regexp_escape.rb

```diff
@@ -71,6 +71,12 @@ def allowed_escape?(node, char, within_character_class)
           end
         end
 
+        def safe_char(char)
```

Comment:
Can you tweak the method name? e.g.

```suggestion
        def enforce_valid_encoding(char)
```

Reply:
Hm, `char.encode('UTF-8', invalid: :replace, undef: :replace)` looks unnecessary.
So it looks like it can be inline like https://github.com/rubocop/rubocop/pull/10906/files#r943095313.